### ISSUE TITLE
[bug] FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD is invalid even if set correctly

### ISSUE THREAD

**Issue by joshuadeguzman** [fastlane/fastlane#17001]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I followed and read the guide on integrating fastlane and uploading the binary to TestFlight via CI. Although [FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD](https://docs.fastlane.tools/best-practices/continuous-integration/#application-specific-passwords) is set, it doesn't seem to be recognized properly by the CI when running `fastlane beta`.

Are there any environment variables I'm missing? I tried setting `FASTLANE_SESSION` as well but that it didn't help, and I don't think it's necessary given `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD` should be enough when skipping 2FA in CI.

CI is Github Actions.

##### Command executed

`fastlane beta`

##### Complete output when running fastlane, including the stack trace and command used

Redacted other logs here.

TL;DR;
prior to this error, everything else ran successfully.

<details>
  <pre> 
(redacted)
[06:26:01]: [Transporter Error Output]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)
[06:26:01]: Transporter transfer failed.
[06:26:01]: 
[06:26:01]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)
[06:26:01]: 
[06:26:01]: Application specific password you provided using
[06:26:01]: environment variable FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD
[06:26:01]: is invalid, please make sure it's correct
  </pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.5                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.5 (19F101)                                      |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.5                                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby

(CREDENTIALS REDACTED)

def delete_temp_keychain(name)
  delete_keychain(
    name: name
  ) if File.exist? File.expand_path("~/Library/Keychains/#{name}-db")
end

def create_temp_keychain(name, password)
  create_keychain(
    name: name,
    password: password,
    unlock: false,
    timeout: false
  )
end

def ensure_temp_keychain(name, password)
  delete_temp_keychain(name)
  create_temp_keychain(name, password)
end

platform :ios do
  lane :development_profiles do
    match(
      type: 'development',
      app_identifier: "#{APP_IDENTIFIER}",
      readonly: false
    )
  end

  lane :appstore_profiles do
    keychain_name = ENV['TEMP_KNAME'] || TEMP_KNAME_DEFAULT
    keychain_password = ENV['TEMP_KPASSWORD'] || TEMP_KPASSWORD_DEFAULT
    ensure_temp_keychain(keychain_name, keychain_password)

    match(
      type: 'appstore',
      app_identifier: "#{APP_IDENTIFIER}",
      git_basic_authorization: Base64.strict_encode64(ENV["GH_ACCESS_TOKEN"]),
      readonly: true,
      keychain_name: keychain_name,
      keychain_password: keychain_password 
    )
  end

  lane :profiles do
    development_profiles
    appstore_profiles
  end

  lane :post_deploy_beta do
    keychain_name = ENV['TEMP_KNAME'] || TEMP_KNAME_DEFAULT
    delete_temp_keychain(keychain_name)
  end

  lane :build_development do
    development_profiles
    gym(
      configuration: "Debug",
      workspace: "Runner.xcworkspace",
      scheme: "Runner"
    )
  end

  xcargs_override = {
    :BUNDLE_IDENTIFIER => APP_ID,
    :PROVISIONING_PROFILE_SPECIFIER => PROFILE,
    :DEVELOPMENT_TEAM => TEAM_ID
  }  
  
  lane :build_beta do
    appstore_profiles
    gym(
      configuration: "Release",
      workspace: "Runner.xcworkspace",
      scheme: "Runner",
      xcargs: xcargs_override,
      export_method: "app-store",
      export_options: {
        provisioningProfiles: { 
            APP_ID => PROFILE
        }
      }
    )
  end

  lane :deploy_beta do
    pilot(
      apple_id: "#{APP_ID}",
      app_identifier: "#{APP_IDENTIFIER}",
      skip_waiting_for_build_processing: true,
      skip_submission: true,
      distribute_external: false,
      notify_external_testers: false,
      ipa: "./Runner.ipa"
    )
  end
  
  lane :beta do
    build_beta
    deploy_beta
    post_deploy_beta
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("REDACTED") # The bundle identifier of your app
apple_id("REDACTED") # Your Apple email address

itc_team_id("REDACTED") # App Store Connect Team ID
team_id("REDACTED") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.154.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.8.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.76.0       |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday                   | 1.0.1        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.10.1       |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.6        |
| bundler                   | 2.1.2        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.2.1        |
| uber                      | 0.1.0        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.15.0       |
| signet                    | 0.14.0       |
| memoist                   | 0.16.2       |
| os                        | 1.1.1        |
| googleauth                | 0.13.1       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.3        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| rake                      | 13.0.1       |
| digest-crc                | 0.6.1        |
| google-cloud-storage      | 1.27.0       |
| emoji_regex               | 3.0.0        |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.350.0      |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.2.1        |
| aws-sdk-core              | 3.104.3      |
| aws-sdk-kms               | 1.36.0       |
| aws-sdk-s3                | 1.75.0       |
| nanaimo                   | 0.3.0        |
| colored2                  | 3.1.2        |
| claide                    | 1.0.3        |
| CFPropertyList            | 3.0.2        |
| atomos                    | 0.1.3        |
| xcodeproj                 | 1.17.1       |
| unicode-display_width     | 1.7.0        |
| public_suffix             | 4.0.5        |
| addressable               | 2.7.0        |
| json                      | 2.3.1        |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.1.0        |
| rexml                     | 3.2.4        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.4.2        |
</details>

*generated on:* **2020-08-06**
</details>

**Comment by joshuadeguzman** [comment]
I solved it by adding single quote, which is very absurd and hard to debug. 

Hope this helps other folks out there:
https://github.community/t/secret-with-special-escape-characters-fails-when-used/17108
